rclpy/rclpy/qos_event.py

```diff
@@ -225,7 +225,7 @@ def __init__(
         self.use_default_callbacks = use_default_callbacks
 
     def create_event_handlers(
-        self, callback_group: CallbackGroup, subscription_handle: Handle,
+        self, callback_group: CallbackGroup, subscription_handle: Handle, topic_name: str,
```

Comment:
I think it's fine to break API, because this is mostly used internally.
It seems weird to add the `topic_name` as an argument only because one of the "default generated" callbacks need it.

Maybe `create_event_handlers` could take a `rclpy.subscription.Subscription` instead of both the `topic_name` and the `subscription_handle`. That seems a bit more flexible if we want to query something else from the `Subscription` in the future.

Reply:
> I think it's fine to break API, because this is mostly used internally.

Yeah, that was my thinking.

> Maybe `create_event_handlers` could take a `rclpy.subscription.Subscription` instead of both the `topic_name` and the `subscription_handle`. That seems a bit more flexible if we want to query something else from the `Subscription` in the future.

The issue there is that we create a circular dependency between subscription.py and qos_event.py.  subscription.py needs to import qos_event.py to do things, and then passing a `subscription.Subscription` into a `SubscriptionEventHandler` means that qos_event.py needs a dependency on subscription.py to get the subscription type.  I think Python will figure this out, and it may even be possible to only have this dependency at type-checking time, but I'd rather not have the circular dependency at all.  Other thoughts welcome, though.

Reply:
> The issue there is that we create a circular dependency between subscription.py and qos_event.py. subscription.py needs to import qos_event.py to do things, and then passing a subscription.Subscription into a SubscriptionEventHandler means that qos_event.py needs a dependency on subscription.py to get the subscription type. I think Python will figure this out, and it may even be possible to only have this dependency at type-checking time, but I'd rather not have the circular dependency at all. Other thoughts welcome, though.

Circular dependencies at type checking time aren't an issue, they can be solved using `typing.TYPE_CHECKING` and quoting the type annotation.
I find the circular dependency "normal" in this case, but feel free to ignore the comment.

Reply:
Thanks for the feedback.  I think I'm going to go ahead with this patch as-is, just to get this in.  If we need to have more information from the Subscription object in the future, we can revisit it then.